services/gitlab/gitlab-issues.service.js

```diff
@@ -0,0 +1,291 @@
+        color: 'yellow',
+      },
+      documentation,
+    },
+    {
+      title: 'GitLab all issues',
+      pattern: 'all-raw/:project+',
+      namedParams: {
+        project: 'gitlab-org/gitlab',
+      },
+      queryParams: { gitlab_url: 'https://gitlab.com' },
+      staticPreview: {
+        label: 'all issues',
+        message: '115k',
+        color: 'yellow',
+      },
+      documentation,
+    },
+    {
+      title: 'GitLab all issues by-label',
+      pattern: 'all-raw/:project+',
+      namedParams: {
+        project: 'gitlab-org/gitlab',
+      },
+      queryParams: {
+        labels: 'test,failure::new',
+        gitlab_url: 'https://gitlab.com',
+      },
+      staticPreview: {
+        label: 'all test,failure::new issues',
+        message: '20',
+        color: 'yellow',
+      },
+      documentation: documentation + labelDocumentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'issues', color: 'informational' }
+
+  static render({ issues }) {
+    return {
+      label: issues.renderLabel,
+      message: issues.renderMsg,
+      color: issues.renderColor,
+    }
+  }
+
+  async fetch({ project, baseUrl, labels }) {
+    // https://docs.gitlab.com/ee/api/issues_statistics.html#get-project-issues-statistics
+    return super.fetch({
+      schema,
+      url: `${baseUrl}/api/v4/projects/${encodeURIComponent(
+        project
+      )}/issues_statistics`,
+      options: labels ? { searchParams: { labels } } : undefined,
+      errorMessages: {
+        404: 'project not found',
+      },
+    })
+  }
+
+  static transform({ variant, raw, statistics, labels }) {
+    const state = variant
+    const isMultiLabel = labels && labels.includes(',')
+    const labelText = labels ? `${isMultiLabel ? `${labels}` : labels} ` : ''
+
+    let labelPrefix = ''
+    let messageSuffix = ''
+    if (raw !== undefined) {
+      labelPrefix = `${state} `
+    } else {
+      messageSuffix = state
+    }
+
+    let issueCount
+    switch (state) {
+      case 'open':
+        issueCount = statistics.counts.opened
+        break
+      case 'closed':
+        issueCount = statistics.counts.closed
+        break
+      case 'all':
+        issueCount = statistics.counts.all
+        break
+    }
+    const renderLabel = `${labelPrefix}${labelText}issues`
+    const renderMsg = `${metric(issueCount)}${
+      messageSuffix ? ' ' : ''
+    }${messageSuffix}`
+    const renderColor = issueCount > 0 ? 'yellow' : 'brightgreen'
+
+    return {
+      renderLabel,
+      renderMsg,
+      renderColor,
+    }
+  }
+
+  async handle(
+    { variant, raw, project },
+    { gitlab_url: baseUrl = 'https://gitlab.com', labels }
+  ) {
+    const { statistics } = await this.fetch({
+      project,
+      baseUrl,
+      labels,
+    })
+    return this.constructor.render({
+      issues: this.constructor.transform({ variant, raw, statistics, labels }),
+    })
+    // return this.constructor.render({ variant, raw, statistics, labels })
```

Comment:
Please remember to remove commented out code too :smile:

Reply:
ops...